src/routes/news/components/NewsList.js

```diff
@@ -0,0 +1,251 @@
+    }
+  };
+
+  renderNoItemsComponent = () => {
+    const { t, theme } = this.props
+    return <NoNews theme={theme}>{t('currentlyNoNews')}</NoNews>
+  };
+
+  rendersNewsListItem = (cityCode: string, language: string) => ({
+    item: newsItem
+  }) => {
+    const { theme, selectedNewsType } = this.props
+    const isTunews = selectedNewsType === TUNEWS
+    return (
+      <NewsListItem
+        key={newsItem.id}
+        newsItem={newsItem}
+        language={language}
+        theme={theme}
+        isTunews={isTunews}
+        navigateToNews={this.navigateToNews(
+          cityCode,
+          language,
+          `${newsItem.id}`
+        )}
+      />
+    )
+  };
+
+  render () {
+    const {
+      newsList,
+      path,
+      cityCode,
+      language,
+      theme,
+      t,
+      fetchMoreNews,
+      isFetchingMore,
+      selectedNewsType
+    } = this.props
+
+    const isTunews = selectedNewsType === TUNEWS
+
+    if (path) {
+      const selectedNewsItem: LocalNewsModel | TunewsModel = newsList.find(
+        _newsItem => `${_newsItem.id}` === path
+      )
+
+      if (selectedNewsItem) {
+        const { eNewsNo, date, content, message } = selectedNewsItem
+
+        return (
+          <View style={{ flex: 1 }}>
+            <ScrollView
+              showsVerticalScrollIndicator={false}
+              contentContainerStyle={{
+                flexGrow: 1,
+                marginBottom: 10,
+                paddingHorizontal: '5%'
+              }}>
+              {isTunews && (
+                <HeaderImageWrapper>
+                  <HeaderImage source={headerImage} />
+                </HeaderImageWrapper>
+              )}
+              <Container>
+                <NewsDetailsTitle theme={theme}>
+                  {selectedNewsItem.title}
+                </NewsDetailsTitle>
+                <NewsDetailsContent theme={theme} language={language}>
+                  {content || message}
+                </NewsDetailsContent>
+              </Container>
+              <Row theme={theme} language={language}>
+                {isTunews ? (
+                  <ExtraInfo theme={theme}>
+                    {`${t('eNewsNo')}: ${eNewsNo}`}
+                    {'  '}
+                    <ExtraInfo onPress={this.openTunewsLink} theme={theme} underlined>tünews INTERNATIONAL</ExtraInfo>
+                    {'  '}
+                    {moment(date).format('MMMM DD, YYYY')}
+                  </ExtraInfo>
+                ) : null}
+              </Row>
+            </ScrollView>
+          </View>
+        )
+      }
+
+      const error = new ContentNotFoundError({
+        type: 'news',
+        id: path,
+        city: cityCode,
+        language
+      })
+      return (
+        <Failure
+          errorMessage={error.message}
+          code={ErrorCodes.PageNotFound}
+          t={t}
+          theme={theme}
+        />
+      )
+    }
+
+    return (
+      <SpaceBetween>
```

Comment:
Why is this necessary? There is only one child component.

Reply:
Right, will delete it.

Reply:
@abeeralshaer